code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/OpenLDAP/OpenLDAPUtils.java

```diff
@@ -14,8 +19,42 @@ public String tokenType() {
 
     @Override
     protected boolean isWhitelisted(List<String> idList) {
-        //TODO Real white listing needed.
-        return true;
+        if (idList == null || idList.isEmpty()) {
+            return false;
+        }
+        List<String> whitelistedValues = fromHashSeparatedString(OpenLDAPConstants.LDAP_ALLOWED_IDENTITIES.get());
+
+        for (String id : idList) {
+            for (String whiteId: whitelistedValues){
+                if (StringUtils.equals(id, whiteId)){
```

Comment:
It is a good question. But we don't have to worry about this though, since the data getting compared "idList" as well as the  "whitelistedValues" saved in our DB are coming from same source - openLDAP directory, so the data will be the same.